torch/lib/c10d/comm.hpp

```diff
@@ -87,6 +87,20 @@ class TORCH_PYTHON_API CommHookInterface {
       const c10::IValue& result) = 0;
 };
 
+// This helper function is called both by CppCommHookInterface below and inside
+// reducer.
+inline std::vector<at::Tensor> _parseCppCommHookResult(const c10::IValue& result) {
```

Comment:
What is the reason for prefixing this function name with an underscore? To mark it as an "internal" function? If that is the case, I suggest moving it to a `detail` namespace instead (common convention in C++).

```c++
namespace detail {
std::vector<at::Tensor> parseCppCommHookResult(const c10::IValue& result) {
  ...
}
}
```

Reply:
Sounds good. I was wonder what's the good programming style in this codebase. It seems that creating a `details` namespace is consisting with other similar scenarios under `torch`.